src/MIDebugEngine/AD7.Impl/AD7BoundBreakpoint.cs

```diff
@@ -54,7 +51,22 @@ public AD7BoundBreakpoint(AD7Engine engine, AD7PendingBreakpoint pendingBreakpoi
         // Called when the breakpoint is being deleted by the user.
         int IDebugBoundBreakpoint2.Delete()
         {
-            return Delete();
+            if (Delete() == Constants.S_OK)
+            {
+                if ((enum_BP_LOCATION_TYPE)PendingBreakpoint.BpRequestInfo.bpLocation.bpLocationType == enum_BP_LOCATION_TYPE.BPLT_DATA_STRING)
+                {
+                    lock (_engine.DebuggedProcess.DataBreakpointVariables)
+                    {
+                        string addressName = HostMarshal.GetDataBreakpointStringForIntPtr(PendingBreakpoint.BpRequestInfo.bpLocation.unionmember3);
```

Comment:
1. See my other note about using `AddressId` instead of `GetDataBreakpointStringForIntPtr`
2. You want to null check that the AddressId before adding it.